timemachine/fe/free_energy.py

```diff
@@ -1459,30 +1504,10 @@ def format_rates(rs):
     assert isinstance(potential, custom_ops.SummedPotential)
     unbound_impls = potential.get_potentials()
 
-    def make_energy_decomposed_state(
-        results: Tuple[StoredArrays, List[NDArray], InitialState],
-    ) -> EnergyDecomposedState[StoredArrays]:
-        frames, boxes, initial_state = results
-        # Reuse the existing unbound potentials already constructed to make a batch Us fn
-        return EnergyDecomposedState(
-            frames,
-            boxes,
-            get_batch_u_fns(unbound_impls, [p.params for p in initial_state.potentials], temperature),
-        )
-
-    results_by_state = [
-        (samples.frames, samples.boxes, initial_state)
-        for samples, initial_state in zip(samples_by_state, initial_states)
-    ]
+    neighbor_ulkns = generate_pair_bar_ulkns(initial_states, samples_by_state, temperature, unbound_impls=unbound_impls)
 
-    bar_results = list(
-        pairwise_transform_and_combine(
-            results_by_state,
-            make_energy_decomposed_state,
-            lambda s1, s2: estimate_free_energy_bar(compute_energy_decomposed_u_kln([s1, s2]), temperature),
-        )
-    )
+    pair_bar_results = [estimate_free_energy_bar(u_kln, temperature) for u_kln in neighbor_ulkns]
```

Comment:
[local naming self-consistency] There might be an error in one or the other name? "`... for u_kln in neighbor_ulkns ...`"

Reply:
Ahh, I just meant that `*_lkns` sounds like it's a list of elements (each `*_lkn`) not transposed elements (each `*_kln`). It's just naming though -- the logic looked fine.